FWCore/Framework/interface/ComponentFactory.h

```diff
@@ -55,31 +55,28 @@ namespace edm {
       std::shared_ptr<base_type> addTo(EventSetupsController& esController,
                                        EventSetupProvider& iProvider,
                                        edm::ParameterSet& iConfiguration,
-                                       ModuleTypeResolverBase const* resolver,
+                                       ModuleTypeResolverMaker const* resolverMaker,
                                        bool replaceExisting = false) const {
         std::string modtype = iConfiguration.template getParameter<std::string>("@module_type");
         //cerr << "Factory: module_type = " << modtype << endl;
         typename MakerMap::iterator it = makers_.find(modtype);
+        Maker* maker = nullptr;
```

Comment:
bare pointer feels `icky`. Maybe use a non owning smart pointer? Also, could this be `const`?

Reply:
pThe cache in `Factory` uses `unique_ptr`, whereas the cache in `ComponentFactory` uses `shared_ptr`, so out-of-the box e.g. `weak_ptr` can't be used. On the other hand, I don't see any reason why `Factory` and `ComponentFactory` could not use the same smart pointer.

So I could move both caches to use `shared_ptr` and use `weak_ptr` here, but given that (IIUC) using the `weak_ptr` would be effectively the same as copying the `shared_ptr`, I wonder if it would really be worth it? (even if the module creation is not that performance critical code). 

Making these pointers to pointers-to-`const` seems to work.